sw/device/lib/base/freestanding/limits.h

```diff
@@ -0,0 +1,51 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_BASE_FREESTANDING_LIMITS_H_
+#define SW_DEVICE_LIB_BASE_FREESTANDING_LIMITS_H_
+
+#include <stdint.h>
+
+/**
+ * This header implements the limits.h standard header, as required by C11 S4.6.
+ * This header is specified in detail in S7.10 and S5.2.4.2.1 of the same.
+ */
+
+#define CHAR_BIT (8)
+#define SCHAR_MIN INT8_MIN
+#define SCHAR_MAX INT8_MAX
+#define UCHAR_MAX UINT8_MAX
+
+// NOTE: S5.2.5.2.2 specifies that the following defines should corresond to the
```

Comment:
Couldn't find S5.2.5.2.2, do you mean S5.2.4.2.1? (I only have a draft of the standard though. Things may be different in the final version.)

Reply:
It should be `S5.2.4.2.1p2` I think. Good catch.

Reply:
You're right. I might have been using an older draft. Though this refers to the second paragraph, so the correct sitation is S5.2.4.2.1p2.